modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -118,40 +118,70 @@ const isInvalidUri = (settings: SaxParserSettings, uri: string, tagName: string)
   }
 };
 
+interface LightweightTag {
+  readonly tagDepth: number;
+  readonly end: number;
+}
+
+// This finds a tag, but with very small amounts of detail - used only in findMatchingEndTagIndex
+// Instead of returning what type of tag it is, we just say "is it an open tag, a close tag, or a self-closing tag (like a br, or comment)"
+// And we also don't bother returning the start index of the tag, just the end
+const findLightweightTag = (html: string, startIndex: number, shortEndedElements: SchemaMap): LightweightTag | null => {
```

Comment:
Another minor issue here is going to be less performance than the old code and parsing is a hot code path. It does a lot of additional scope switching and object creation for the return values. As shown recently by the Boulder work that does make a difference for hot code paths, especially when it starts triggering GC collections (which is also related to the memory issue and why why shouldn't be using slice).

As such, did this really need to be split out? I don't believe so and am assuming it was only done for readability.